### ISSUE TITLE
Is scraping shadow DOM an option?

### ISSUE THREAD

**Issue by jtlimo** [geziyor/geziyor#48]
Hi, I'm trying to web scrapping [YouTube charts](https://charts.youtube.com/?hl=pt), unsuccessfully because they use polymer / shadow DOM. With Geziyor, could I do that? I'm using colly, and they don't have support for that.

**Comment by musabgultekin** [comment]
Its not possible rn. Since actions are hardcoded. Im working on adding a way to add custom actions.

**Comment by musabgultekin** [comment]
Ive added custom actions in request.
So you'll add custom chromedp.Action to the geziyor.Request object. like req.Actions = []chromedp.Actions{XXX}

See TestGetRenderedCustomActions test

https://github.com/geziyor/geziyor/commit/738852f9321de26c193ae88a9b2fb4d6aebb6540

And you'll access to shadow doms like this action:

chromedp.Click(`document.querySelector('#container').shadowRoot.querySelector('#foo')`, chromedp.ByJSPath),

See this issue: https://github.com/chromedp/chromedp/issues/376

Let me know if this fixes your problem!

**Comment by jtlimo** [comment]
> Ive added custom actions in request. So you'll add custom chromedp.Action to the geziyor.Request object. like req.Actions = []chromedp.Actions{XXX}
> 
> See TestGetRenderedCustomActions test
> 
> [738852f](https://github.com/geziyor/geziyor/commit/738852f9321de26c193ae88a9b2fb4d6aebb6540)
> 
> And you'll access to shadow doms like this action:
> 
> chromedp.Click(`document.querySelector('#container').shadowRoot.querySelector('#foo')`, chromedp.ByJSPath),
> 
> See this issue: [chromedp/chromedp#376](https://github.com/chromedp/chromedp/issues/376)
> 
> Let me know if this fixes your problem!

I'll try this between today and tomorrow… after testing, I say if it is work for my case.

**Comment by musabgultekin** [comment]
Yeah I can look into it tomorrow, busy for today. Will comment here when i can find time.
